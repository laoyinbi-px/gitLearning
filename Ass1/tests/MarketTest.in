add trader1 100
add trader2 100
add trader3 100
TRADES TRADER trader1
import trader1 ABC 100
ORDER 0000
buy trader2 ABC 10 7.5
ORDER 0003
ORDER 0000
SELL trader1 ABC 15 5
buy trader3 ABC 10 7.5
SELL trader1 ABC 1000 5
BOOK BUY
CANCEL BUY 0002
TRADES PRODUCT ABC
exit