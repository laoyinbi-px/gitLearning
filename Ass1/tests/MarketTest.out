$ Success.
$ Success.
$ Success.
$ No trades have been completed by trader.
$ Trader now has 100.00 units of ABC.
$ No orders in either book in the market.
$ No trades could be made, order added to buy book.
$ Order is not present in either order book.
$ 0000: BUY 10.00xABC @ $7.50
$ Product sold in part, trades as follows:
trader1->trader2: 10.00xABC for $7.50.
$ Product bought in part, trades as follows:
trader1->trader3: 5.00xABC for $5.00.
$ Order could not be placed onto the market.
$ 0002: BUY 5.00xABC @ $7.50
$ Order successfully cancelled.
$ trader1->trader2: 10.00xABC for $7.50.
trader1->trader3: 5.00xABC for $5.00.
$ Have a nice day.

